src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\Persistence\ConnectionRegistry;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    /**
+     * @param \ArrayObject<int, array{name: string, timing: int}> $connectionTiming
+     * @param array<string, int>                                  $timeToLiveByConnection
+     */
+    public function __construct(
+        private readonly \ArrayObject $connectionTiming,
+        private ConnectionRegistry $connectionRegistry,
+        private readonly array $timeToLiveByConnection,
```

Comment:
I think this concern (the by-connection aspect) should live in the driveror the middleware.
We can remove this map if we use $connectionExpiries instead of $connectionTiming (aka the timestamp where the object is EOLed)

Reply:
But I'm not sure I understand: don't we need a reference point to compare timings and decide whether to close or not?

Reply:
I think I understand what you mean.